### PR TITLE
chore: update react-intl and formatjs deps, add eslint plugin for formatjs

### PR DESCRIPTION
**Pull request by emmenko**
Just found out about this: https://formatjs.io/docs/tooling/cli#builtin-formatters

Also Transifex supports structured JSON: https://docs.transifex.com/formats/json/structured-json

### CODE REVIEW

File: .eslintrc.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ env:
   node: true
   mocha: false
 plugins:
+  - 'formatjs'
```
Comment (emmenko) [inline .eslintrc.yaml:22]: They also provide an (experimental) eslint plugin

File: packages/components/fields/money-field/src/messages.js [hunk 2]
```diff
@@ -1,9 +0,0 @@
-import { defineMessages } from 'react-intl';
-
-export default defineMessages({
-  highPrecision: {
```
Comment (emmenko) [inline packages/components/fields/money-field/src/messages.js:4]: This is duplicated in the `money-input` component, and is not used in the money field.

```
warning [FormatJS CLI] Duplicate message id: "UIKit.MoneyInput.highPrecision", but the `description` and/or `defaultMessage` are different.
```

**Comment by emmenko** [comment]
We'll see 🙂

**Comment by emmenko** [comment]
We needed to change the file format in the tx integration:

image

**Comment by tdeekens** [comment]
> We'll see 🙂

UIKit is a small project and as a result a good candidate. We have a backup in the pulled things we could merge into the busted things I guess. Should we inform the translation team that we're trying this?

**Comment by tdeekens** [comment]
> We'll see 🙂

UIKit is a small project and as a result a good candidate. We have a backup in the pulled things we could merge into the busted things I guess. Should we inform the translation team that we're trying this?

**Comment by emmenko** [comment]
> Should we inform the translation team that we're trying this?

Yep, already did

> UIKit is a small project and as a result a good candidate. 

I agree, we can try this here and see how it goes.

**Comment by emmenko** [comment]
It might be that we need to adjust how we load the translated messages, depending on how transifex converts them. And also compile the messages: https://formatjs.io/docs/getting-started/message-distribution#translation-management-system-tms-integration

**Comment by emmenko** [comment]
This is how the compiled output should look like (I think):

image

**Comment by emmenko** [comment]
FYI: I changed it back to `KEYVALUEJSON` for now until @ikumiot can get the feedback if this is useful or not.

**Pull request merged**